controller_manager/test/test_controller_manager_srvs.cpp

```diff
@@ -189,6 +193,117 @@ TEST_F(TestControllerManagerSrvs, list_controllers_srv)
+  ASSERT_EQ(result->controller[1].required_command_interfaces.size(), 0u);
+  ASSERT_EQ(result->controller[1].required_state_interfaces.size(), 0u);
+  ASSERT_EQ(result->controller[1].is_chainable, false);
+  ASSERT_EQ(result->controller[1].is_chained, false);
+  ASSERT_EQ(result->controller[1].reference_interfaces.size(), 0u);
+  ASSERT_EQ(result->controller[1].chain_connections.size(), 0u);
+  // configure controllers
+  cm_->configure_controller(test_chainable_controller::TEST_CONTROLLER_NAME);
+  cm_->configure_controller(test_controller::TEST_CONTROLLER_NAME);
+  // get controller list after configure
+  result = call_service_and_wait(*client, request, srv_executor);
+  ASSERT_EQ(2u, result->controller.size());
+  // check chainable controller
+  ASSERT_EQ(result->controller[0].state, "inactive");
+  ASSERT_EQ(result->controller[0].claimed_interfaces.size(), 0u);
+  ASSERT_EQ(result->controller[0].required_command_interfaces.size(), 1u);
+  ASSERT_EQ(result->controller[0].required_state_interfaces.size(), 2u);
+  ASSERT_EQ(result->controller[0].is_chainable, true);
+  ASSERT_EQ(result->controller[0].is_chained, false);
+  ASSERT_EQ(result->controller[0].reference_interfaces.size(), 2u);
+  ;
+  ASSERT_EQ(result->controller[0].chain_connections.size(), 0u);
+  // check test controller
+  ASSERT_EQ(result->controller[1].state, "inactive");
+  ASSERT_EQ(result->controller[1].claimed_interfaces.size(), 0u);
+  ASSERT_EQ(result->controller[1].required_command_interfaces.size(), 3u);
+  ASSERT_EQ(result->controller[1].required_state_interfaces.size(), 2u);
+  ASSERT_EQ(result->controller[1].is_chainable, false);
+  ASSERT_EQ(result->controller[1].is_chained, false);
+  ASSERT_EQ(result->controller[1].reference_interfaces.size(), 0u);
+  ASSERT_EQ(result->controller[1].chain_connections.size(), 1u);
+  // activate controllers
+  cm_->switch_controller(
+    {test_chainable_controller::TEST_CONTROLLER_NAME}, {},
+    controller_manager_msgs::srv::SwitchController::Request::STRICT, true, rclcpp::Duration(0, 0));
+  cm_->switch_controller(
+    {test_controller::TEST_CONTROLLER_NAME}, {},
+    controller_manager_msgs::srv::SwitchController::Request::STRICT, true, rclcpp::Duration(0, 0));
+  // get controller list after activate
+  result = call_service_and_wait(*client, request, srv_executor);
+  // check chainable controller
+  ASSERT_EQ(result->controller[0].state, "active");
+  ASSERT_EQ(result->controller[0].claimed_interfaces.size(), 1u);
+  ASSERT_EQ(result->controller[0].is_chained, true);
+  // check test controller
+  ASSERT_EQ(result->controller[1].state, "active");
+  ASSERT_EQ(result->controller[1].claimed_interfaces.size(), 3u);
+  ASSERT_EQ(
+    test_chainable_controller::TEST_CONTROLLER_NAME,
+    result->controller[1].chain_connections[0].name);
+  ASSERT_EQ(2u, result->controller[1].chain_connections[0].reference_interfaces.size());
+  ASSERT_EQ("joint1/position", result->controller[1].chain_connections[0].reference_interfaces[0]);
```

Comment:
Shouldn't interface names actually be:
`std::string(test_chainable_controller::TEST_CONTROLLER_NAME) + "/joint1/position"`?

Reply:
Ok, I see this also make sense in this case... This is good, thanks!